Sources/PenguinStructures/NumberOperations.swift

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension BinaryInteger {
+  /// Returns a collection of the positive integers that are coprime with `self`.
+  ///
+  /// Definition: Two numbers are coprime if their GCD is 1.
+  public var positiveCoprimes: PositiveCoprimes<Self> { .init(self) }
+
+  /// Returns positive integers that are coprime with, and less than, `self`.
+  ///
+  /// - SeeAlso: `positiveCoprimes`.
+  public var lesserPositiveCoprimes: [Self] {
+    return positiveCoprimes.prefix { $0 < self }
+  }
+}
+
+/// The positive values that are coprime with *N*.
+///
+/// Example:
+/// ```
+/// print(Array(10.positiveCoprimes.prefix(8)))  // [1, 3, 7, 9, 11, 13, 17, 19]
+/// ```
+///
+/// Note: Although there are infinitely many positive prime numbers, `PositiveCoprimes` is bounded
+/// by the maximum representable integer in `Domain` (if such a limit exists, such as the
+/// `FixedWidthInteger` types).
+public struct PositiveCoprimes<Domain: BinaryInteger>: Collection {
+  // TODO: Specialize SubSequence for efficiency.
+
+  /// The number to find coprimes relative to.
+  public let target: Domain
+
+  /// The index into the collection of positive coprimes are the coprimes themselves.
+  ///
+  /// Note: the indices are not dense or contiguous in `Domain`.
+  public typealias Index = Domain
+
+  /// Creates a collection of numbers coprime relative to `target`.
+  internal init(_ target: Domain) {
+    self.target = target < 0 ? target * -1 : target
+  }
+
+  /// `Int.max`, as there are infinitely many prime numbers, and thus infinitely many coprimes
+  /// to a given target.
+  public var count: Int {
+    if _slowPath(target == 0) { return 0 }
+    return Int.max
+  }
+
+  /// Accesses the coprime at `index`.
+  public subscript(index: Index) -> Domain {
+    index
+  }
+
+  /// The first valid coprime (1).
```

Comment:
```suggestion
  /// The index of the first coprime (1).
```

I'm not sure it's any better to write that than

```suggestion
  /// The index of the first element.
```

Think through how this will show up in the user/reader experience before you document generic requirements in more specific ways.